mobile/src/screens/HomeScreen.jsx

```diff
@@ -1,15 +1,40 @@
-import { Button, Text } from "react-native"
+import { Button, FlatList, ScrollView, Text } from "react-native"
 import Page from "../components/Page"
 import screens from "../utils/constants/screens"
+import { getCategories } from "../utils/categoriesServices"
+import { useEffect, useState } from "react"
+import ErrorScreen from "./ErrorScreen"
+import errors from "../utils/constants/errors"
 
 const HomeScreen = props => {
   const { navigation } = props
-  const goToCategory = () => navigation.navigate(screens.categoryPage)
+  const [categories, setCategories] = useState([])
+  const [hadError, setHadError] = useState(false)
```

Comment:
You're using a useEffect, so you should be using present perfect. 
Something like : ` const [hasError, setError] = useState(false)`

- You're setting an error and something has an error